ethcore/snapshot/src/watcher.rs

```diff
@@ -104,20 +102,36 @@ impl Watcher {
 
 impl ChainNotify for Watcher {
 	fn new_blocks(&self, new_blocks: NewBlocks) {
-		if self.oracle.is_major_importing() || new_blocks.has_more_blocks_to_import { return }
-
-		trace!(target: "snapshot_watcher", "{} imported", new_blocks.imported.len());
+		if self.oracle.is_major_importing() {
+			trace!(target: "snapshot_watcher", "Is major importing; not taking snapshot");
+		} else if new_blocks.has_more_blocks_to_import {
+			trace!(target: "snapshot_watcher", "Has more blocks to import; not taking snapshot");
+			return
+		}
 
+		// Decide if it's time for a snapshot: the highest of the imported blocks is a multiple of 5000?
 		let highest = new_blocks.imported.into_iter()
+			// Convert block hashes to block numbers for all newly imported blocks
 			.filter_map(|h| self.oracle.to_number(h))
+			// …only keep the new blocks that have numbers bigger than period + history
+			// todo:    this seems nonsensical: period is always 5000 and history is always 100, so
+			//          this filters out blocknumbers that are lower than 5100; what's the point of that?
 			.filter(|&num| num >= self.period + self.history)
+			// Subtract `history` (i.e. `SNAPSHOT_HISTORY`, i.e. 100) from the block numbers.
+			// todo:    why? Here we back off 100 blocks such that the final block
+			//          number from which we start the snapshot is "(new) highest block" - 100)
+			//          Maybe we do this to avoid IO contention on the tip? Or for reorgs? If the
+			//          latter, it should be maybe be higher?
```

Comment:
my bet is on reorgs, but maybe @rphmeier or @arkpar could comment on that

Reply:
yes, it's about reorgs.